stix/idl/io/stx_map2fits.pro

```diff
@@ -0,0 +1,285 @@
+;+
+; Project     : STIX
+;
+; Name        : STX_MAP2FITS
+;
+; Purpose     : Write STIX image reconstructed map to FITS file. The header of
+;               newly created FITS file depends on the reconstruction algorithm
+;               used.
+;
+; Category    : imaging
+;
+; Syntax      : stx_map2fits,map,file,path_sci_file
+;
+; Inputs      : MAP = image map structure
+;               FILE = FITS file name
+;               PATH_SCI_FILE = path to the L1 file used for reconstructing the STIX image
+;
+; Keywords    : PATH_BKG_FILE = path to the L1 BKG file included for the reconstruction of the STIX image
+;               XY_SHIFT = applied shift to the location of the map. Provide the aspect if applied
+;               ERR = error string
+;               BYTE_SCALE = byte scale data
+;               ALL_CLEAN = store all maps of the clean map structure 
+;                           (by default only the clean map is stored, i.e., clean_map[0])
+;
+; Comments    : Based on map2fits in sswidl, but adapted for STIX images
+; History     : 05-04-2022, Hualin Xiao (hualin.xiao@fhnw.ch)
+;                - copied map2fits from ssw and add more keywords to the 
+;                  header to make generated fits files be compatible with sunpy.map
+;                
+;                - 17.05.2022: A. F. Battaglia (andrea-battaglia@fhnw.ch)
+;                  It now includes most of the keywords already present in the L1 science file.
+;                  Now the user MUST specify path_sci_file
+;
+;                - 04.07.2022: A. F. Battaglia (andrea-battaglia@fhnw.ch)
+;                  Keyword ALL_CLEAN added. By default, only the clean_map[0] is stored
+;
+;                - 04.11.2022: A. F. Battaglia
+;                  DATE-OBS added (before only DATE_OBS). For better compatibility with
+;                  other languages.
+;                  
+;                - 03.04.2023: A. F. Battaglia
+;                  RSUN_REF added. The same value as in the EUI FITS files has been added
+; -
+
+   pro stx_map2fits,in_map,file,path_sci_file,path_bkg_file=path_bkg_file,$
+    xy_shift=xy_shift, err=err,byte_scale=byte_scale,verbose=verbose,all_clean=all_clean
+
+   err=''
+   verbose=keyword_set(verbose)
+
+  ; ******************** ADAPTED FOR STIX ************************
+
+  ; Check if the L1 file is given
+  if not keyword_set(path_sci_file) then begin
+    print, ''
+    print, ' >>>>>>>>>> path_sci_file has to be defined!! <<<<<<<<<<'
+    print, ''
+    message, 'Please, define path_sci_file (path to the science file used for imaging)'
+  endif
+    
+  ; Check if the algorithm used is CLEAN.
+  ; If this is the case, then store only the CLEAN map and discard all other maps
```

Comment:
Could include the other images

Reply:
Yes, couldn't other maps be included in other extensions?  What would `sunpy.map.Map` do in that situation?

Reply:
Pretty sure it just ignores any additional image present

Reply:
Sure, I'll do it!